editor/post-title/index.js

```diff
@@ -88,12 +121,15 @@ class PostTitle extends Component {
 
 	render() {
 		const { title } = this.props;
-		const { isSelected } = this.state;
-		const className = classnames( 'editor-post-title', { 'is-selected': isSelected } );
+		const { isSelected, hasFocusWithin } = this.state;
+		const className = classnames( 'editor-post-title', { 'is-selected': isSelected && hasFocusWithin } );
 
 		return (
-			<div className={ className }>
-				{ isSelected && <PostPermalink /> }
+			<div
+				className={ className }
+				onBlur={ this.onOuterBlur }
+				onFocus={ this.onOuterFocus }>
+				{ isSelected && hasFocusWithin && <PostPermalink onLinkCopied={ this.focusText } /> }
```

Comment:
You mentioned the clipboard button library being the cause that led to this implementation to focus text again after copying. Is that something that can be (or perhaps is already) fixed at the library? Or fixed on the clipboard button? Maybe using our "withFocusReturn" higher-order component could work as a solution?

https://github.com/WordPress/gutenberg/blob/master/components/higher-order/with-focus-return/index.js

Reply:
I think that might work, but now that I think about it, doesn't this only work on componentDidUnmount? The only react controlled component is the copy button (the textarea that gets focus is created by the clipboard library itself), and I thought the copy button wasn't being removed from the DOM after clicking on it?

Reply:
There is also an option in the clipboard library of passing through trigger, and then calling clearSelection (which will focus trigger and then remove the selection), but it would be just calling focus outside of the react in the exact same way. Is that actually preferable?